operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/BulkDataClient.java

```diff
@@ -652,53 +771,81 @@ public String submitImport(String inputFormat, String inputSource, List<Input> i
 
         String fhirTenant = FHIRRequestContext.get().getTenantId();
         builder.fhirTenant(fhirTenant);
 
+        builder.outcome(outcome);
+
+        builder.source(source);
+
         String fhirDataStoreId = FHIRRequestContext.get().getDataStoreId();
         builder.fhirDataStoreId(fhirDataStoreId);
 
         String entityStr = JobInstanceRequest.Writer.generate(builder.build(), true);
-        Entity<String> entity = Entity.json(entityStr);
-        Response r = target.request().post(entity);
-
-        // TODO check the HTTP response status code
-        // e.g. if it comes back with a 4xx or a 500 it may fail on the JobInstanceResponse.Parser.parse!
-        String responseStr = r.readEntity(String.class);
 
-        // Debug / Dev only
+        String baseUrl = adapter.getCoreApiBatchUrl() + "/jobinstances";
         if (log.isLoggable(Level.FINE)) {
-            log.warning("$import response (HTTP " + r.getStatus() + ") -> \n" + responseStr);
+            log.fine("The Base URL " + baseUrl);
+            log.fine("The Entity posted to the server " + entityStr);
         }
 
-        JobInstanceResponse response = JobInstanceResponse.Parser.parse(responseStr);
+        CloseableHttpClient cli = wrapper.getHttpClient(adapter.getCoreApiBatchUser(), adapter.getCoreApiBatchPassword());
+        HttpPost jobPost = new HttpPost(baseUrl);
+        StringEntity entity = new StringEntity(entityStr, ContentType.create("plain/text", Consts.UTF_8));
+        jobPost.setEntity(entity);
+        CloseableHttpResponse jobResponse = cli.execute(jobPost);
+
+        int status = -1;
+        String jobId = "-1";
+        try {
+            status = jobResponse.getStatusLine().getStatusCode();
+            handleStandardResponseStatus(status);
+
+            // Debug / Dev only
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("$import response (HTTP " + status + ")");
+            }
+
+            if (status != 201) {
+                // Job is not created
+                throw export.buildOperationException("Unable to create the $import job", IssueType.INVALID);
+            }
+
+            String responseString = new BasicResponseHandler().handleResponse(jobResponse);
+
+            JobInstanceResponse response = JobInstanceResponse.Parser.parse(responseString);
 
-        // From the response
-        String jobId = Integer.toString(response.getInstanceId());
+            jobId = Integer.toString(response.getInstanceId());
 
-        String baseUri = properties.get(BulkDataConfigUtil.BASE_URI);
-        return baseUri + "/$bulkdata-status?job="
-                + BulkDataExportUtil.encryptBatchJobId(jobId, BulkDataConstants.BATCHJOBID_ENCRYPTION_KEY);
+        } finally {
+            jobPost.releaseConnection();
+            jobResponse.close();
+        }
+        cli.close();
+
+        return baseUri + "/$bulkdata-status?job=" + transformer.endcodeJobId(jobId);
     }
 
-    /**
+    /*
```

Comment:
why not javadoc?

Reply:
it really isn't intended  too be publicaly readable.  It's also a private method.  thus the `/*`

Reply:
same comment as the other one...I think its totally acceptable to use javadoc-style comments on private methods

Reply:
```suggestion
    /**
```